### PR TITLE
Propose Bryce Buchanan as approver of the project

### PR DESCRIPTION
**Pull request by nachoBonafonte**
I wanted to propose Bryce Buchanan ( @bryce-b ) as an approver of the project.

* He has been a reviewer of the codebase for several months now
* He is the author of more than 10 PR's and has helped reviewing many others
* He is regularly attending Swift SIG meeting and is an active member of the otel-swift Slack channel

This PR adds him to the CODEOWNERS file (I am not sure if this is the correct way of adding him though)

### CODE REVIEW

File: CODEOWNERS
Comment (Sherlouk) [inline CODEOWNERS:15]:
```suggestion
# Maintainers @SergeyKanzhelev @nachoBonafonte
```

**Review by Sherlouk — APPROVED** [review]
Minor change before merge, but I support this change.

**Comment by SergeyKanzhelev** [comment]
I support this change. 

A few comments on the technicality are you proposing @bryce-b as approver or maintainer? Codeowners suggests "maintainer" nomination. Perhaps we can maintain the list of approvers and maintainers in readme for differentiation like this: https://github.com/open-telemetry/opentelemetry-dotnet#contributing. Also I like the structure of CODEOWNERS here: https://github.com/open-telemetry/opentelemetry-go/blob/main/CODEOWNERS where it's clear that maintainers must approve CODEOWNERS

Once this approved, please make sure to add @bryce-b to the team @open-telemetry/swift-approvers

**Review by SergeyKanzhelev — APPROVED** [review]
I support. I left a comments on better structure declaring approvers and maintainers

**Comment by nachoBonafonte** [comment]
I have changed to the opentelemetry-go idea, is really clever

**Comment by nachoBonafonte** [comment]
@SergeyKanzhelev I could not add @bryce-b to the @open-telemetry/swift-approvers, it says "not a member of this organization"

**Comment by SergeyKanzhelev** [comment]
@bryce-b can you please file [membership issue](https://github.com/open-telemetry/community/issues/new?template=membership.md&title=REQUEST%3A%20New%20membership%20for%20%3Cyour-GH-handle%3E) as per [instruction](https://github.com/open-telemetry/community/blob/main/community-membership.md#:~:text=open%20an%20issue%20against%20the%20opentelemetry%2Fcommunity%20repo)?

**Comment by bryce-b** [comment]
@SergeyKanzhelev  done! https://github.com/open-telemetry/community/issues/746

**Pull request merged**
